ivre/db/mongo.py

```diff
@@ -721,6 +721,53 @@ def getscreenshot(port):
         if url == "field":
             return port.get('screendata')
 
+    def set_label(self, flt, group, label, archives=False):
+        """Adds `label` in `group` to every host matching `flt`"""
+        colname = self.colname_oldhosts if archives else self.colname_hosts
+        for host in self.find(colname, flt):
+            labels = host.setdefault('labels', [])
```

Comment:
Since you do not use `host['labels']` after that, you can replace `.setdefault()` by `.get()`.